src/pocketmine/inventory/PlayerInventory.php

```diff
@@ -453,7 +453,7 @@ public function sendSlot($index, $target){
 	 * @return Human|Player
 	 */
 	public function getHolder(){
-		return parent::getHolder();
+		return $this->holder;
```

Comment:
There are already other instances of these overrides. They should not affect performance, and they are indeed necessary for IDEs to identify the type (for autocompletion) in a loosely typed language like PHP.

My change only carries out premature optimization to the method, and does not really change anything that can be worse.